Plugin/Catalog/Model/Config.php

```diff
@@ -0,0 +1,113 @@
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace  Nosto\Tagging\Plugin\Catalog\Model;
+
+use Magento\Catalog\Model\Config as MagentoConfig;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Magento\Backend\Block\Template\Context;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Framework\View\Element\Template;
+
+class Config extends Template
+{
+    const NOSTO_PERSONALIZED_KEY = 'nosto-personalized';
+
+    const NOSTO_TOPLIST_KEY = 'nosto-toplist';
+
+    const NONE = 'none';
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var NostoHelperAccount */
+    private $nostoHelperAccount;
+
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    public $nostoSortingOptions = [
+        self::NOSTO_PERSONALIZED_KEY => 'Personalized for you',
+        self::NOSTO_TOPLIST_KEY => 'Top products'
+    ];
+
+    /**
+     * Config constructor.
+     * @param NostoHelperData $nostoHelperData
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param Context $context
+     * @param StoreManagerInterface $storeManager
+     * @param array $data
+     */
+    public function __construct(
+        NostoHelperData $nostoHelperData,
+        NostoHelperAccount $nostoHelperAccount,
+        Context $context,
+        StoreManagerInterface $storeManager,
+        array $data = []
+    ) {
+        $this->nostoHelperData = $nostoHelperData;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->storeManager = $storeManager;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Add custom Sorting attribute
+     *
+     * @param MagentoConfig $catalogConfig
+     * @param $options
+     * @return array
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
```

Comment:
Missing `* @throws \Magento\Framework\Exception\NoSuchEntityException` that can be thrown in `$store = $this->storeManager->getStore();`